### PR TITLE
pkg/expect: ExpectFunc, LineCount

### PR DESCRIPTION
**Pull request by gyuho**
ExpectFunc to make expect more extensible. LineCount to be
able to check 'no output' command.

### CODE REVIEW

**Comment by gyuho** [comment]
/cc @heyitsanthony 

Thanks.

File: pkg/expect/expect_test.go [hunk 1]
```diff
@@ -18,6 +18,24 @@ package expect
 
 import "testing"
 
+func TestExpectFunc(t *testing.T) {
+	ep, err := NewExpect("/bin/echo", "hello world")
+	if err != nil {
+		t.Fatal(err)
+	}
+	wstr := "hello world\r\n"
+	l, eerr := ep.ExpectFunc(func(a string) bool { return wstr == a })
```
Comment (heyitsanthony) [inline pkg/expect/expect_test.go]: this is too much like `Expect`... maybe `return len(a) > 10`?

File: pkg/expect/expect.go [hunk 2]
```diff
@@ -95,6 +97,20 @@ func (ep *ExpectProcess) Expect(s string) (string, error) {
...
+func (ep *ExpectProcess) Expect(s string) (string, error) {
+	return ep.ExpectFunc(func(txt string) bool { return strings.Contains(txt, s) })
+}
+
+// LineCount returns the number of recorded lines since
+// the beginning of the process.
+func (ep *ExpectProcess) LineCount() int {
+	ep.mu.Lock()
+	defer ep.mu.Unlock()
+
```
Comment (heyitsanthony) [inline pkg/expect/expect.go]: delete this blank line? this sort of function shows up everywhere so no need to space it out

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
thanks. merging in greens.

**Pull request merged**